codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessageDecoder.java

```diff
@@ -339,7 +336,7 @@ private ByteBuf decodeLine(ChannelHandlerContext ctx, ByteBuf buffer) throws Exc
                 discardedBytes = 0;
                 discarding = false;
             } else {
-                discardedBytes = buffer.readableBytes();
+                discardedBytes += buffer.readableBytes();
                 buffer.skipBytes(discardedBytes);
```

Comment:
I think you also need. to change this to `buffer.skipBytes(buffer.readableBytes())` as otherwise you may produce an `IndexOutOfBoundsException`.

Reply:
You're right. Or it would be more efficient to use `buffer.readerIndex(buffer.writerIndex())` instead

Reply:
Its basically the same. :)

Reply:
Done